File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/GetAccountInfo.java
package l2s.gameserver.network.authcomm.as2gs;

import io.netty.buffer.ByteBuf;
import l2s.commons.dbutils.DbUtils;
import l2s.gameserver.Config;
import l2s.gameserver.GameServer;
import l2s.gameserver.database.DatabaseFactory;
import l2s.gameserver.network.authcomm.ReceivablePacket;
import l2s.gameserver.network.authcomm.gs2as.SetAccountInfo;
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class GetAccountInfo extends ReceivablePacket {
    private final String _account;

    public GetAccountInfo(AuthServerConnection client, ByteBuf byteBuf) {
        super(client, byteBuf);

        _account = readString(byteBuf);
    }

    @Override
    protected void runImpl(AuthServerConnection client) {
        int playerSize = 0;
        List<Integer> deleteChars = Collections.emptyList();
        Connection con = null;
        PreparedStatement statement = null;
        ResultSet rset = null;
        try {
            con = DatabaseFactory.getInstance().getConnection();
            statement = con.prepareStatement("SELECT deletetime FROM characters WHERE account_name=?");
            statement.setString(1, _account);
            rset = statement.executeQuery();
            while (rset.next()) {
                ++playerSize;
                int d = rset.getInt("deletetime");
                if (d > 0) {
                    if (deleteChars.isEmpty())
                        deleteChars = new ArrayList<>(3);
                    deleteChars.add(d + Config.CHARACTER_DELETE_AFTER_HOURS * 60 * 60);
                }
            }
        } catch (Exception e) {
            LOGGER.error("GetAccountInfo:run()", e);
        } finally {
            DbUtils.closeQuietly(con, statement, rset);
        }

        GameServer.getInstance().getAuthServerCommunication().sendPacket(new SetAccountInfo(_account, playerSize, deleteChars));
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/GetAccountInfo.kt
package l2s.gameserver.network.authcomm.as2gs

import io.netty.buffer.ByteBuf
import l2s.commons.dbutils.DbUtils
import l2s.gameserver.Config
import l2s.gameserver.database.DatabaseFactory
import l2s.gameserver.network.authcomm.ReceivablePacket
import l2s.gameserver.network.authcomm.gs2as.SetAccountInfo
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection
import org.napile.primitive.Containers
import org.napile.primitive.lists.IntList
import org.napile.primitive.lists.impl.ArrayIntList
import org.slf4j.LoggerFactory
import java.sql.Connection
import java.sql.PreparedStatement
import java.sql.ResultSet

/**
 * @author VISTALL
 * @date 21:05/25.03.2011
 */
class GetAccountInfo(client: AuthServerConnection, byteBuf: ByteBuf) : ReceivablePacket(client, byteBuf) {

    private val _account: String = readString(byteBuf)

    override fun runImpl(client: AuthServerConnection) {
        var playerSize = 0
        var deleteChars: IntList? = Containers.EMPTY_INT_LIST
        var con: Connection? = null
        var statement: PreparedStatement? = null
        var rset: ResultSet? = null
        try {
            con = DatabaseFactory.getInstance().connection
            statement = con.prepareStatement("SELECT deletetime FROM characters WHERE account_name=?")
            statement.setString(1, _account)
            rset = statement.executeQuery()
            while (rset.next()) {
                playerSize++
                val d = rset.getInt("deletetime")
                if (d > 0) {
                    if (deleteChars!!.isEmpty) deleteChars = ArrayIntList(3)
                    deleteChars.add(d + Config.CHARACTER_DELETE_AFTER_HOURS * 60 * 60)
                }
            }
        } catch (e: Exception) {
            _log!!.error("GetAccountInfo:runImpl():$e", e)
        } finally {
            DbUtils.closeQuietly(con, statement, rset)
        }

        client.sendPacket(SetAccountInfo(_account, playerSize, deleteChars!!.toArray()))
    }

    companion object {
        private val _log = LoggerFactory.getLogger(GetAccountInfo::class.java)
    }
}